src/panels/calendar/ha-panel-calendar.ts

```diff
@@ -0,0 +1,304 @@
+  "fb8072",
+  "80b1d3",
+  "fdb462",
+  "fccde5",
+  "bc80bd",
+  "ffed6f",
+  "c4eaff",
+  "cf8c00",
+  "1b9e77",
+  "d95f02",
+  "e7298a",
+  "e6ab02",
+  "a6761d",
+  "0097ff",
+  "00d067",
+  "f43600",
+  "4ba93b",
+  "5779bb",
+  "927acc",
+  "97ee3f",
+  "bf3947",
+  "9f5b00",
+  "f48758",
+  "8caed6",
+  "f2b94f",
+  "eff26e",
+  "e43872",
+  "d9b100",
+  "9d7a00",
+  "698cff",
+  "d9d9d9",
+  "00d27e",
+  "d06800",
+  "009f82",
+  "c49200",
+  "cbe8ff",
+  "fecddf",
+  "c27eb6",
+  "8cd2ce",
+  "c4b8d9",
+  "f883b0",
+  "a49100",
+  "f48800",
+  "27d0df",
+  "a04a9b",
+];
+
+@customElement("ha-panel-calendar")
+class PanelCalendar extends LitElement {
+  @property() public hass!: HomeAssistant;
+
+  @property({ type: Boolean, reflect: true, attribute: "narrow" })
+  public narrow = false;
+
+  @property() private _calendars: SelectedCalendar[] = [];
+
+  @property() private _events: CalendarEvent[] = [];
+
+  private _start?: Date;
+
+  private _end?: Date;
+
+  private _firstUpdate = false;
+
+  protected firstUpdated(changedProps) {
+    super.firstUpdated(changedProps);
+    if (!this.hass) {
+      return;
+    }
+
+    this._fetchCalendars();
+  }
+
+  protected updated() {
+    if (!this._firstUpdate) {
+      this._fetchData();
+    }
+  }
+
+  protected render(): TemplateResult {
+    return html`
+      <app-header-layout has-scrolling-region>
+        <app-header fixed slot="header"
+          ><app-toolbar>
+            <ha-menu-button
+              .hass=${this.hass}
+              .narrow=${this.narrow}
+            ></ha-menu-button>
+            <div main-title>${this.hass.localize("panel.calendar")}</div>
+          </app-toolbar>
+        </app-header>
+        <div class="content">
+          <div class="calendar-list">
+            <ha-card header="Calendars">
+              ${this._calendars.map(
+                (selCal) =>
+                  html`<div class="calendar-toggle">
+                    <paper-checkbox
+                      style=${styleMap({
+                        "--primary-color": selCal.backgroundColor,
+                      })}
+                      .value=${selCal.calendar.entity_id}
+                      .checked=${selCal.selected}
+                      @change=${this._handleToggle}
+                    >
+                      ${selCal.calendar.name}</paper-checkbox
+                    >
+                  </div>`
+              )}
+            </ha-card>
+          </div>
+          <ha-full-calendar
+            .events=${this._events}
+            @view-changed=${this._handleViewChanged}
+          ></ha-full-calendar>
+        </div>
+      </app-header-layout>
+    `;
+  }
+
+  private _fetchCalendars() {
+    this.hass.callApi<Calendar[]>("GET", "calendars").then((result) => {
+      this._calendars = result.map((cal, idx) => ({
+        selected: true,
+        calendar: cal,
+        backgroundColor: `#${palette[idx]}`,
+      }));
+    });
+  }
+
+  private _fetchData() {
```

Comment:
In my head, I think I can make this better. Aka, if the view changes from Month to week. There is no need to re-fetch the events. We have them already. Unless the date range is larger or greater than the current dates of events we have already fetch then we shouldn't need to re-fetch.

My only roadblock is what if they go one day or week past the current dates. Do we delete the data we have and re-fetch or just append to the event array. Is there a limit of events we should hold in memory?

Reply:
If anyone has thoughts on the best practice here. Do we store all the events that are requested? Meaning the start and end just keep growing further apart as you navigate back and forth. if you go back a month it sets the start and the end stays but only fetches the month before to the original start. and same with the end but opposite ways

Reply:
Let's postpone that optimization to after the MVP